### PR TITLE
Refactor dl1 writing

### PR DESCRIPTION
**Pull request by maxnoe**
* Move writer setup / excludes out of the event loop
* Write images before parameter calculations

### CODE REVIEW

File: lstchain/reco/r0_to_dl1.py [hunk 1]
```diff
@@ -381,18 +416,12 @@ def r0_to_dl1(
...
 
-                tel = event.dl1.tel[telescope_id]
-                tel.prefix = ''  # don't really need one
-                # remove the first part of the tel_name which is the type 'LST', 'MST' or 'SST'
-                tel_name = str(subarray.tel[telescope_id])[4:]
 
                 if custom_calibration:
                     lst_calibration(event, telescope_id)
 
-                write_event = True
```
Comment (rlopezcoto) [inline lstchain/reco/r0_to_dl1.py:392]: what was this `write_event` doing and why is it now removed?

File: lstchain/reco/r0_to_dl1.py [hunk 2]
```diff
@@ -369,7 +392,19 @@ def r0_to_dl1(
             # a candidate muon ring. In that case the full image could be kept, or reduced
             # only after the ring analysis is complete.
 
-            for ii, telescope_id in enumerate(event.dl1.tel.keys()):
+            for telescope_id, dl1_tel in event.dl1.tel.items():
+                dl1_tel.prefix = ''  # don't really need one
```
Comment (rlopezcoto) [inline lstchain/reco/r0_to_dl1.py:396]: what is the inizialization of `dl1_tel.prefix`?

File: lstchain/reco/r0_to_dl1.py (hunk 1 above)
Comment (maxnoe) [reply lstchain/reco/r0_to_dl1.py:392]: It was unused.

File: lstchain/reco/r0_to_dl1.py (hunk 2 above)
Comment (maxnoe) [reply lstchain/reco/r0_to_dl1.py:396]: `dl1_camera` created from the name of the container class :`DL1CameraContainer`. But this is not a change, the code is just at a new position.

File: lstchain/reco/r0_to_dl1.py [hunk 3]
```diff
@@ -381,18 +416,12 @@ def r0_to_dl1(
 
                 dl1_container.fill_event_info(event)
 
-                tel = event.dl1.tel[telescope_id]
-                tel.prefix = ''  # don't really need one
```
Comment (maxnoe) [inline lstchain/reco/r0_to_dl1.py:385]: Here is where it happened before @rlopezcoto

**Review by rlopezcoto — APPROVED** [review]

**Pull request merged**